code/modules/reagents/reagent_containers/cups/drinks.dm

```diff
@@ -443,6 +450,57 @@
 		desc += " It has an emblazoned Nanotrasen logo on it."
 		icon_state = "shaker_n"
 
+/obj/item/reagent_containers/cup/glass/shaker/examine(mob/user)
+	. = ..()
+	. += span_notice("Alt-click to [using_custom_drinks?"disable":"enable"] custom drink naming")
+	if(using_custom_drinks)
+		. += span_notice("Drinks poured from this shaker will have the following name: [custom_drink_name]")
+		. += span_notice("Drinks poured from this shaker will have the following description: [custom_drink_desc]")
+
+/obj/item/reagent_containers/cup/glass/shaker/AltClick(mob/user)
+	. = ..()
+	using_custom_drinks = !using_custom_drinks
+
```

Comment:
Note you're not checking that a living mob is doing this, so ghosts can do it. 
Neither are you checking the mob is in a valid state to do it, so they can do this while dead, or from across the room.